File: 119. Pascal's Triangle II/main.py
"""
Author: Amal Matty Antony
"""
from math import factorial
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        """
        n C r = n! / (r! x (n-r)!)
        
        """
        n = rowIndex+1
        n_fac = factorial(n)
        a = 1
        b = n_fac
        
        ans = [1]
        for i in range(1,rowIndex):
            a *= i
            b = b//(n-i)
            print(n_fac, a, b)
            ans.append(n_fac//(a*b))
        return ans

File: 119. Pascal's Triangle II/main.cpp
// Solution By Vandit Jain (jainvandit99)
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> cache = {1,1};
        if(rowIndex == 0) return vector<int>{1};
        int i = 2;
        while(i<=rowIndex){
            vector<int> newRow(i+1,1);
            for(int j = 1; j<=i/2; j++)
                newRow[j] = newRow[i-j] = cache[j-1] + cache[j];
            cache = newRow;
            i++;
        }
        return cache;
    }
};